codesandbox/codesandbox-client
# Add freezing/unfreezing in the dashboard

An addition is the refactoring of our logic when it comes to updating/removing dashboard sandboxes in state. We had duplicated it quite a bit, so I simplified the logic and deduplicated it in a separate `internalActions` file.

Additional changes:

* Deduplicated changing/removing sandbox from state
* Removed need to provide `page` to sandbox change (we now always check REPOS and change it if need be)
* Removed need to provide `oldTitle` to `renameSandbox` (is now inferred in action)
* Reuse our `sandboxFragmentDashboard` in `repoFragmentDashboard` and `templateFragmentDashboard`. This simplifies typing and both had 100% overlap with `sandboxFragmentDashboard`, also when something needs to be added to `sandboxFragmentDashboard` we probably need it in `repoFragmentDashboard` and `templateFragmentDashboard`.
* Slightly changed order of SandboxMenu and MultiMenu items (shown in screenshots below)
* Added margin to eye (request from @DannyRuchtie)
* Added frozen status to sandboxes that are frozen, with a tooltip
* Added possibility using context menu to freeze/unfreeze sandboxes
* Allow everyone with write permission and pro to change privacy of the sandbox in a workspace (not only the author)

Tested:

* Verify changing state works by renaming and changing privacy of a sandbox
* Verify rollback of changing state works by renaming sandbox while it fails
* Verify deletion from state works by deleting the sandbox
* Verify it changes state everywhere, by renaming sandbox in "Recents" and see if the change propagated in the folder
* Verify whether changing frozen works by (un)freezing sandboxes both in multi and single context menu
* Verify whether rollback of changing frozen works by manually throwing error


<img width="203" alt="Screenshot 2020-09-02 at 16 56 55" src="https://user-images.githubusercontent.com/587016/92002892-a929cc00-ed40-11ea-8859-d6911405c7f2.png">
<img width="244" alt="Screenshot 2020-09-02 at 16 56 51" src="https://user-images.githubusercontent.com/587016/92002898-ab8c2600-ed40-11ea-8d3e-0f92b597e3ad.png">

![image](https://user-images.githubusercontent.com/587016/92004397-77b20000-ed42-11ea-90ef-ba595e12d640.png)
